From 405ef98ec560b5ba972cbcce1437f8ec0ae3bc42 Mon Sep 17 00:00:00 2001
From: Matthew Heon <mheon@redhat.com>
Date: Thu, 17 Jul 2014 12:00:51 -0400
Subject: [PATCH] Update minimal secrets to not error on dangling symlinks

Docker-DCO-1.1-Signed-off-by: Matthew Heon <mheon@redhat.com> (github: mheon)
---
 daemon/secrets.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/daemon/secrets.go b/daemon/secrets.go
index a363bf3..8bd97a5 100644
--- a/daemon/secrets.go
+++ b/daemon/secrets.go
@@ -45,6 +45,11 @@ func readAll(root, prefix string) ([]SecretData, error) {
 	for _, f := range files {
 		fileData, err := readFile(root, filepath.Join(prefix, f.Name()))
 		if err != nil {
+			// If the file did not exist, might be a dangling symlink
+			// Ignore the error
+			if os.IsNotExist(err) {
+				continue
+			}
 			return nil, err
 		}
 		data = append(data, fileData...)
-- 
1.9.3
